### PR TITLE
Feature/console

### PR DESCRIPTION
**Pull request by willgarrett64**

### CODE REVIEW

File: src/components/server/ServerConsole.vue [hunk 1]
```diff
@@ -1,41 +1,61 @@
...
       <h4>Server Console</h4>
       <p class="mt-3 text-gray-500">
         Use the server console for native-like terminal access to your server from your browser.
       </p>
-      <a target="_blank" :href="`/server/${server._key}/vnc`" class="h-full mt-4 lg:mt-0 button button--success">
+      <!-- <a target="_blank" :href="`/server/${server._key}/vnc`" class="h-full mt-4 lg:mt-0 button button--success">
         <span>Launch console in new window</span>
         <DuplicateIcon class="w-4 h-4 ml-2" />
-      </a>
+      </a> -->
```
Comment (adamkdean) [inline src/components/server/ServerConsole.vue:11]: If not used, let's clean this up.

File: src/main.js [hunk 2]
```diff
@@ -2,16 +2,15 @@
 // Use of this source code is governed by a GNU GPL-style license
 // that can be found in the LICENSE.md file. All rights reserved.
 
+import './index.css'
+import './utils'
 import App from './App.vue'
+/* Set up input masking for the account number */
```
Comment (adamkdean) [inline src/main.js:8]: Do we need a comment in the imports?

File: src/main.js [hunk 3]
```diff
@@ -2,16 +2,15 @@
...
 import router from './router'
 import store from './store'
 import titleMixin from './mixins/titleMixin'
-import './index.css'
-import './utils'
-
-/* Set up input masking for the account number */
-import { VueMaskDirective } from 'v-mask'
 
 const vMaskV2 = VueMaskDirective
```
Comment (adamkdean) [inline src/main.js:15]: Wouldn't the comment be here?

File: src/views/Vnc.vue [hunk 4]
```diff
@@ -1,80 +1,120 @@
...
+      this.statusClass = 'noVNC_status_normal'
+      this.status = `Connected to ${this.server.settings.name}`
       this.rfb.focus()
     },
+    disconnected() {
+      setTimeout(() => {
+        this.statusClass = 'noVNC_status_error'
+        this.status = `Failed to connect to ${this.server.settings.name}`
+        this.close(true)
+      }, 1000)
```
Comment (adamkdean) [inline src/views/Vnc.vue:55]: 1000 ms may be a bit too fine. I suggest a slightly longer timeout, perhaps 5000 ms.

**Review by adamkdean — CHANGES_REQUESTED** [review]
LGTM pending a few small changes

File: src/account-utils/servers.js [hunk 5]
```diff
@@ -113,6 +113,14 @@ export const getTasks = async (host, sessionId, serverId, params) => {
   return response.body
 }
 
+// get a server's VNC credentials, including a session id and VNC password
+export const getVncCredentials = async (host, sessionId, serverId) => {
```
Comment (annybs) [inline src/account-utils/servers.js:117]: Should this be renamed to `getVncSession` in line with the naming in API?

File: src/components/server/ServerConsole.vue (hunk 1 above)
Comment (annybs) [reply src/components/server/ServerConsole.vue:11]: +1

File: src/main.js (hunk 2 above)
Comment (annybs) [reply src/main.js:8]: +1

File: src/main.js (hunk 3 above)
Comment (annybs) [reply src/main.js:15]: +1

File: src/views/Vnc.vue (hunk 4 above)
Comment (annybs) [reply src/views/Vnc.vue:55]: +1

**Pull request merged**